src/hooks/useWebSocket/useWebSocket.js

```diff
@@ -1,35 +1,36 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import io from "socket.io-client";
 
-export const useWebSocket = ({ url, onOpen, onMessage, onClose, onError }) => {
-  useEffect(() => {
-    const websocket = new WebSocket(url);
-
-    if (onOpen) {
-      websocket.addEventListener("open", (event) => {
-        onOpen(event, websocket);
-      });
-    }
-
-    if (onMessage) {
-      websocket.addEventListener("message", (event) => {
-        onMessage(event, websocket);
-      });
-    }
+export const useWebSocket = ({ url }) => {
+  const socket = io(url);
```

Comment:
We might want to keep this in useState. This will create a new socket reference every render.

Reply:
ah yes!  ok, useRef probably even better for this case.   thx